koku/masu/processor/tasks.py

```diff
@@ -494,3 +511,99 @@ def autovacuum_tune_schema(schema_name):
+)
+def convert_to_parquet(request_id, account, provider_uuid, provider_type, start_date, manifest_id, context={}):
+    """
+    Convert archived CSV data from our S3 bucket for a given provider to Parquet.
+
+    This function chiefly follows the download of a providers data.
+
+    This task is defined to attempt up to 10 retries using exponential backoff
+    starting with a 10-second delay. This is intended to allow graceful handling
+    of temporary AWS S3 connectivity issues because it is relatively important
+    for us to convert the archived data.
+
+    Args:
+        request_id (str): The associated request id (ingress or celery task id)
+        account (str): The account string
+        provider_uuid (UUID): The provider UUID
+        start_date (str): The report start time (YYYY-mm-dd)
+        manifest_id (str): The identifier for the report manifest
+        context (dict): A context object for logging
+
+    """
+    if not context:
+        context = {"account": account, "provider_uuid": provider_uuid}
+
+    if not request_id or not account or not provider_uuid:
+        if not request_id:
+            message = "missing required argument: request_id"
+            LOG.error(message)
+        if not account:
+            message = "missing required argument: account"
+            LOG.error(message)
+        if not provider_uuid:
+            message = "missing required argument: provider_uuid"
+            LOG.error(message)
+        if not provider_type:
+            message = "missing required argument: provider_type"
+            LOG.error(message)
+        return
+
+    if not settings.ENABLE_S3_ARCHIVING:
+        msg = "Skipping convert_to_parquet. S3 archiving feature is disabled."
+        LOG.info(log_json(request_id, msg, context))
+        return
+
+    if not start_date:
+        msg = "S3 archiving feature is enabled, but no start_date was given for processing."
+        LOG.warn(log_json(request_id, msg, context))
+        return
+
+    try:
+        cost_date = datetime.datetime.strptime(start_date, "%Y-%m-%d")
+    except ValueError:
+        msg = "S3 archiving feature is enabled, but the start_date was not a valid date string format (YYYY-mm-dd)."
+        LOG.warn(log_json(request_id, msg, context))
+        return
+
+    s3_csv_path = get_path_prefix(account, provider_uuid, cost_date, Config.CSV_DATA_TYPE)
+    local_path = f"{Config.TMP_DIR}/{account}/{provider_uuid}"
+    s3_parquet_path = get_path_prefix(account, provider_uuid, cost_date, Config.PARQUET_DATA_TYPE)
+
+    file_keys = get_file_keys_from_s3_with_manifest_id(request_id, s3_csv_path, manifest_id, context)
+    files = [os.path.basename(file_key) for file_key in file_keys]
+    if not files:
+        msg = "S3 archiving feature is enabled, but no files to process."
+        LOG.info(log_json(request_id, msg, context))
+        return
+
+    if provider_type == Provider.PROVIDER_AWS or provider_type == Provider.PROVIDER_AWS_LOCAL:
```

Comment:
Why is this restricted to just AWS?  Would we ever want to clear OCP or Azure files?

Reply:
So you saying this plus Andrew's pause in a previous conversation made me go back and look at Azure too (thought it was only spitting out data for that data, but its daily data for the month). Looks like both Azure/AWS reports are the entire month, so we only want the data from the current report so we can remove the previously stored data as we are processing the new batch. OCP data provide daily granularity. So this should be updated to included Azure too.

Reply:
Think you have OCP and Azure reversed maybe?  Azure is daily.

Reply:
Nope. not reversed. The data in Azure is daily values for the month (report contains multiple days). With the division of OCP data into daily chunked CSVs here: https://github.com/project-koku/koku/pull/2210/files#diff-6918b2e09ba9007181c66df408cc6072R132-R157

Reply:
Logic updated here.